pkgs/tools/graphics/directx-shader-compiler/default.nix

```diff
@@ -0,0 +1,50 @@
+{ stdenv, fetchFromGitHub, cmake, python3, git }:
+
+stdenv.mkDerivation rec {
+  pname = "directx-shader-compiler";
+  version = "1.5.2010";
+
+  # Put headers in dev, there are lot of them which aren't necessary for
+  # using the compiler binary.
+  outputs = [ "out" "dev" ];
+
+  src = fetchFromGitHub {
+    owner = "Microsoft";
+    repo = "DirectXShaderCompiler";
+    rev = "b640fa4ba4afb8eb9f15b4a34ab5187c7d94de97";
+    sha256 = "0ccfy1bfp0cm0pq63ri4yl1sr3fdn1a526bsnakg4bl6z4fwrnnj";
+    fetchSubmodules = true;
+  };
+
+  nativeBuildInputs = [ cmake git python3 ];
```

Comment:
Why is git required? Does it need to fetch something else or does it parse a version number?

Reply:
Yeah, it puts `rev-parse HEAD` into the build somewhere.

Reply:
Can you patch that and hardcode the version number? IIRC if fetchGit is used without submodules there is no .git directory.

Reply:
Sure, but is it worth the time?

I added a note to the `fetchSubmodules` line mentioning that we rely on the side effect of fetching the .git directory.